pkg/providers/instanceprofile/instanceprofile.go

```diff
@@ -129,5 +148,55 @@ func (p *DefaultProvider) Delete(ctx context.Context, instanceProfileName string
 		return awserrors.IgnoreNotFound(serrors.Wrap(fmt.Errorf("deleting instance profile, %w", err), "instance-profile", instanceProfileName))
 	}
 	p.cache.Delete(instanceProfileName)
+	p.protectedProfiles.Delete(instanceProfileName)
 	return nil
 }
+
+func (p *DefaultProvider) ListClusterProfiles(ctx context.Context) ([]*iamtypes.InstanceProfile, error) {
```

Comment:
We should consider caching these calls in the future too - I'd consider it out of scope of this PR though since these calls are rate limited by the controllers making them.